spec/syntax/parser_spec.cr

```diff
@@ -2474,8 +2474,29 @@ describe "Parser" do
     fn
       ->(1, :hi) { true || false }
     end
-  ),                        AnonymousFunction.new([Block.new([p(nil, l(1)), p(nil, l(:hi))], e(Or.new(l(true), l(false))))])
+  ),                          AnonymousFunction.new([Block.new([p(nil, l(1)), p(nil, l(:hi))], e(Or.new(l(true), l(false))))])
 
+  # Exception handling is not allowed with {... } syntax, but only with do... end
+  it_does_not_parse %q(
+    fn
+      ->() {
+        raise :error
+        rescue
+          :rescued
+      }
+    end
+  )
+
+  # Expection handling in anonymous functions with do... end syntax is allowed
+  it_parses %q(
+    fn
+      ->() do
+        raise :error
+        rescue
+          :rescue
+        end
+    end
+  ),                        AnonymousFunction.new([Block.new(body: ExceptionHandler.new(Raise.new(l(:error)), [Rescue.new(l(:rescue))]))])
 
```

Comment:
It'd be nice to cover more cases here as well: rescue with parameters, multiple rescues, rescue with ensure, etc. These may fit better with the rest of the [exception handling specs](https://github.com/myst-lang/myst/blob/master/spec/syntax/parser_spec.cr#L1687-L2056). In fact, just duplicating all of the specs with `def` and replacing it with `fn ->() do` would probably do the trick.

Reply:
Its quite a lot, and is it not `parse_exception_handler` which is being used in the exception handling specs? 

That is just the same which is used in the anonymous functions, I mean, I have no experience in this, so i just wonder why this is different.

Reply:
The goal is to have complete coverage of the syntax, without any assumptions about how the parser is implemented. So, while the parser is currently using `parse_exception_handler` in both cases, that isn't _necessarily_ guaranteed. It's also possible that a future change to `parse_exception_handler` could add some stateful condition that causes it to work properly on exception handlers for blocks, but not anonymous functions, etc. Without tests for all the possibilities on anonymous functions, we wouldn't be able to catch that immediately.

It's certainly tedious to start with, but the complete test coverage means we can much more confidently make larger changes to the parser without worry of missing or breaking something.

Reply:
Gotcha! I'll see to it